nifi-nar-bundles/nifi-groovyx-bundle/nifi-groovyx-processors/src/test/java/org/apache/nifi/processors/groovyx/ExecuteGroovyScriptTest.java

```diff
@@ -427,6 +427,29 @@ public void test_withWriter() throws Exception {
         flowFile.assertContentEquals("5678".getBytes(StandardCharsets.UTF_16LE));
     }
 
+    @Test
+    public void test_onStart_onStop() throws Exception {
+        runner.setProperty(ExecuteGroovyScript.SCRIPT_FILE, TEST_RESOURCE_LOCATION + "test_onStart_onStop.groovy");
+        runner.assertValid();
+        runner.enqueue("");
+        runner.run();
+
+        runner.assertAllFlowFilesTransferred(ExecuteGroovyScript.REL_SUCCESS.getName(), 1);
+        final List<MockFlowFile> result = runner.getFlowFilesForRelationship(ExecuteGroovyScript.REL_SUCCESS.getName());
+        MockFlowFile resultFile = result.get(0);
+        resultFile.assertAttributeExists("a");
+        resultFile.assertAttributeEquals("a", "A");
+        // Inspect the output visually for onStop, no way to pass back values
+    }
+
+    @Test
```

Comment:
If you'd like to actually check the out here you can do something like this.
```
@Test
public void test_onUnscheduled() throws Exception {
    runner.setProperty(ExecuteGroovyScript.SCRIPT_FILE, TEST_RESOURCE_LOCATION + "test_onUnscheduled.groovy");
    runner.assertValid();
    final PrintStream originalOut = System.out;
    final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
    System.setOut(new PrintStream(outContent));
    runner.run();
    System.setOut(originalOut);
    assertEquals("onUnscheduled invoked successfully - Inspect this visually\n", outContent.toString());
    // Inspect the output visually
}
```

Reply:
Sounds good, I'll give this a go, better to have the computer validate than the eyes :) Thanks!